src/Bicep.LangServer/BicepParamsCompilationManager.cs

```diff
@@ -35,7 +48,25 @@ public void RefreshCompilation(DocumentUri uri, bool reloadBicepConfig = false)
 
         public void UpsertCompilation(DocumentUri uri, int? version, string text, string? languageId = null)
         {
-            var semanticModel = new ParamsSemanticModel(SourceFileFactory.CreateBicepParamFile(uri.ToUri(), text));
+
+            var paramsFile = SourceFileFactory.CreateBicepParamFile(uri.ToUri(), text);
+            var semanticModel = new ParamsSemanticModel(paramsFile);
+
+            var usingDeclarations = paramsFile.ProgramSyntax.Children.OfType<UsingDeclarationSyntax>();
+            var bicepFilePath = PathHelper.TryGetUsingPath(usingDeclarations.SingleOrDefault(), out var getUsingPathFailureBuilder);
+
+            if(bicepFilePath == null)
+            {
+                if(!Uri.TryCreate(uri.ToUri(), bicepFilePath, out var bicepFileUri))
+                {
+                    if(bicepFileUri is {})
```

Comment:
If `Uri.TryCreate` returns `true`, then `bicepFileUri` will be non-null. I believe if that method returns `false`, `bicepFileUri` will always be null.